app/ui/base/src/main/java/com/fsck/k9/ui/base/ThemeManager.kt

```diff
@@ -5,37 +5,43 @@ import android.content.res.Configuration
 import android.os.Build
 import androidx.annotation.StyleRes
 import androidx.appcompat.app.AppCompatDelegate
-import com.fsck.k9.K9
-import com.fsck.k9.K9.AppTheme
-import com.fsck.k9.K9.SubTheme
+import com.fsck.k9.preferences.AppTheme
+import com.fsck.k9.preferences.GeneralSettingsManager
+import com.fsck.k9.preferences.SubTheme
+import kotlinx.coroutines.GlobalScope
+import kotlinx.coroutines.flow.distinctUntilChanged
+import kotlinx.coroutines.flow.launchIn
+import kotlinx.coroutines.flow.map
+import kotlinx.coroutines.flow.onEach
 
 class ThemeManager(
     private val context: Context,
-    private val themeProvider: ThemeProvider
+    private val themeProvider: ThemeProvider,
+    private val generalSettingsManager: GeneralSettingsManager
 ) {
     val appTheme: Theme
-        get() = when (K9.appTheme) {
+        get() = when (generalSettingsManager.getSettings().appTheme) {
```

Comment:
I think this would read nicer if you extracted `generalSettingsManager.getSettings()` to a private property.

```kotlin
private val generalSettings: GeneralSettings
    get() = generalSettingsManager.getSettings()
```